blocks/extras/algorithms/__init__.py

```diff
@@ -0,0 +1,67 @@
+from blocks.algorithms import CompositeRule, Scale, StepRule
+from blocks.utils import shared_floatx
+
+
+class BasicNesterovMomentum(StepRule):
+    u"""Accumulates step with exponential discount.
+
+    Parameters
+    ----------
+    momentum : float, optional
+        The momentum coefficient. Defaults to 0.
+
+    Notes
+    -----
+    This step rule is intended to be used in conjunction with another
+    step rule, _e.g._ :class:`Scale`. For an all-batteries-included
+    experience, look at :class:`NesterovMomentum`.
+
+    It is implemented as in [BBR13].
+
+    .. [BBR13] Yoshua Bengio, Nicolas Boulanger-Lewandowski, and Razvan
+        Pascanu, *Advances in optimizing recurrent networks*,
+        ICASSP (2013), pp 8624-8628.
```

Comment:
Can you also add links to original papers? Seems, that it was introduced in Nesterov, Y.  A method of solving a convex programming problem with convergence rate O (1/k2)